src/tests/test_bigint.cpp

```diff
@@ -774,6 +773,57 @@ class DSA_ParamGen_Test final : public Text_Based_Test {
 
 BOTAN_REGISTER_TEST("math", "dsa_param", DSA_ParamGen_Test);
 
+std::vector<Test::Result> test_bigint_serialization() {
+   auto rng = Test::new_rng("test_bigint_serialization");
+
+   return {
+      Botan_Tests::CHECK("BigInt binary serialization",
+                         [](Test::Result& res) {
+                            Botan::BigInt a(0x1234567890ABCDEF);
+                            auto enc = Botan::BigInt::encode(a);
+                            res.test_eq("BigInt::encode()", enc, Botan::hex_decode("1234567890ABCDEF"));
+
```

Comment:
Maybe, while you're at it:
```suggestion
                            Botan::BigInt a(0x1234567890ABCDEF);
                            auto enc = Botan::BigInt::encode(a);
                            res.test_eq("BigInt::encode()", enc, Botan::hex_decode("1234567890ABCDEF"));

                            auto enc_locked = Botan::BigInt::encode_locked(a);
                            res.test_eq(
                               "BigInt::encode_locked()", enc_locked, Botan::hex_decode_locked("1234567890ABCDEF"));

```